### DISCUSSION TITLE
How to set the bg color for a custom component?

### DISCUSSION THREAD

**Discussion by simonmclean** [nvim-lualine/lualine.nvim#700] [Q&A] [answered]
I have this function which I'd like to use as a custom component

```lua
local function lsp_statusline_errors()
  local count = #vim.diagnostic.get(0, { severity = vim.diagnostic.severity.ERROR })
  if (count > 0) then
    return count
  end
  return ''
end
```

And I'm setting it in `sections.lualine_c` like this

```lua
      {
        lsp_statusline_errors,
        { color = { bg = 'DiagnosticError' } }
      },
```

But no matter how I try to set the color, nothing seems to work. It always just shows the value returned by `lsp_statusline_errors`, without any coloring

**Comment by fitrh** [comment, marked as answer]
> :help lualine-General-component-options

The color option accept the following values

- string as highlight group
- table in the form `{ fg = number|string, bg = number|string, gui = string }`
- function that returns string or table as before

So if you want to define `bg`, you have to provide a hex string or a number

```lua
color = { bg = "#ff0000" }
```

If you want to use a specific highlight group, then the value must be a string

```lua
color = "DiagnosticError"
```


Each component have this structure

```lua
{
    function,
    color = {}, -- or string or number
    -- ...
}
```

**Reply by simonmclean** [reply]
I've tried a few different things, but none seem to work

```lua
      {
        lsp_statusline_errors,
        { color = 'DiagnosticError' }
      },
```

```lua
      {
        lsp_statusline_errors,
        { color = { bg = '#4287f5' } }
      },
```

```lua
      {
        lsp_statusline_errors,
        { color = { bg = '#4287f5', fg = '#4287f5', gui = '#4287f5' } }
      },
```

With all the above the result is the same - The value is displayed plainly with no color.

Here's [my config](https://github.com/simonmclean/neovim-config/blob/master/lua/plugin-configs/lualine.lua) if that helps

**Reply by fitrh** [reply]
`color` is a key in the component table, your component should look like this

```lua
{
    component_function,
    color = {},
    -- ...
}
```

**Reply by simonmclean** [reply]
That's working now. Thanks for the help!
